server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -303,7 +303,7 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
         assert shardSize >= 0 : shardSize;
         long freeBytesAfterShard = freeBytes - shardSize;
         if (freeBytesAfterShard < diskThresholdSettings.getFreeBytesThresholdHighStage(total).getBytes()) {
-            logger.warn(
+            logger.debug(
```

Comment:
Was wondering if should wrap the call with `if (logger.isDebugEnabled()) {` to avoid unnecessary computations in production.

Reply:
If you do not mind I would go with the  simplest option now (`logger.isDebugEnabled()` similar to other places in this decider) and will do the rest in the followups